File: clog-go/db/entity/service-log-entity.go
/*
 * Copyright (c) 2023.
 * Created by Andy Pangaribuan. All Rights Reserved.
 *
 * This product is protected by copyright and distributed under
 * licenses restricting copying, distribution and decompilation.
 */

package entity

import "time"

type ServiceLog struct {
	Id         string    `db:"id"`
	Uid        string    `db:"uid"`
	UserId     *string   `db:"user_id"`
	PartnerId  *string   `db:"partner_id"`
	Xid        *string   `db:"xid"`
	SvcName    string    `db:"svc_name"`
	SvcVersion string    `db:"svc_version"`
	SvcParent  *string   `db:"svc_parent"`
	Endpoint   string    `db:"endpoint"`
	Version    string    `db:"version"`
	Message    *string   `db:"message"`
	Severity   string    `db:"severity"`
	Path       string    `db:"path"`
	Function   string    `db:"function"`
	ReqHeader  *string   `db:"req_header"`
	ReqBody    *string   `db:"req_body"`
	ReqPar     *string   `db:"req_par"`
	ResData    *string   `db:"res_data"`
	ResCode    *int      `db:"res_code"`
	Data       *string   `db:"data"`
	Error      *string   `db:"error"`
	StackTrace *string   `db:"stack_trace"`
	ClientIp   string    `db:"client_ip"`
	DurationMs int       `db:"duration_ms"`
	StartAt    time.Time `db:"start_at"`
	FinishAt   time.Time `db:"finish_at"`
	CreatedAt  time.Time `db:"created_at"`
}

File: clog-dart/lib/db/entity/service_log_entity.dart
/*
 * Copyright (c) 2022.
 * Created by Andy Pangaribuan. All Rights Reserved.
 *
 * This product is protected by copyright and distributed under
 * licenses restricting copying, distribution and decompilation.
 */

class ServiceLogEntity {
  String id;
  String uid;
  String? userId;
  String? partnerId;
  String? xid;
  String svcName;
  String svcVersion;
  String? svcParent;
  String endpoint;
  String version;
  String? message;
  String severity; // types: info, warning, error
  String path;
  String function;
  String? reqHeader;
  String? reqBody;
  String? reqPar;
  String? resData;
  int? resCode;
  String? data;
  String? error;
  String? stackTrace;
  String clientIP;
  late int durationMs;
  DateTime startAt;
  DateTime finishAt;
  DateTime createdAt;

  ServiceLogEntity({
    required this.id,
    required this.uid,
    this.userId,
    this.partnerId,
    this.xid,
    required this.svcName,
    required this.svcVersion,
    this.svcParent,
    required this.endpoint,
    required this.version,
    this.message,
    required this.severity,
    required this.path,
    required this.function,
    this.reqHeader,
    this.reqBody,
    this.reqPar,
    this.resData,
    this.resCode,
    this.data,
    this.error,
    this.stackTrace,
    required this.clientIP,
    required this.startAt,
    required this.finishAt,
    required this.createdAt,
  }) {
    durationMs = finishAt.difference(startAt).inMilliseconds;
  }
}
